modules/apps/callbacks/types/events.go

```diff
@@ -0,0 +1,97 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+)
+
+const (
+	// EventTypeSourceCallback is the event type for a source callback
+	EventTypeSourceCallback = "ibc_src_callback"
+	// EventTypeDestinationCallback is the event type for a destination callback
+	EventTypeDestinationCallback = "ibc_dest_callback"
+
+	// AttributeKeyCallbackTrigger denotes the condition that the callback is executed on:
+	//   "acknowledgement": the callback is executed on the acknowledgement of the packet
+	//   "timeout": the callback is executed on the timeout of the packet
+	//   "recv_packet": the callback is executed on the reception of the packet
+	AttributeKeyCallbackTrigger = "callback_trigger"
+	// AttributeKeySourceCallbackAddress denotes the source callback contract address
+	AttributeKeySourceCallbackAddress = "src_callback_address"
+	// AttributeKeyDestCallbackAddress denotes the destination callback contract address
+	AttributeKeyDestCallbackAddress = "dest_callback_address"
+	// AttributeKeyCallbackResult denotes the callback result:
+	//   "success": the callback is successfully executed
+	//   "failure": the callback is failed to execute
+	AttributeKeyCallbackResult = "callback_result"
+	// AttributeKeyCallbackError denotes the callback error message
+	// if no error is returned, then this key will not be included in the event
+	AttributeKeyCallbackError = "callback_error"
+	// AttributeKeyCallbackGasLimit denotes the custom gas limit for the callback execution
+	// if custom gas limit is not in effect, then this key will not be included in the event
+	AttributeKeyCallbackGasLimit = "callback_gas_limit"
+	// AttributeKeyCallbackPortID denotes the port ID of the packet
+	AttributeKeyCallbackPortID = "callback_port"
+	// AttributeKeyCallbackChannelID denotes the channel ID of the packet
+	AttributeKeyCallbackChannelID = "callback_channel"
```

Comment:
Should we specify in the key name that these are for the **source** port ID and channel ID?

Reply:
done